libmscore/measure.cpp

```diff
@@ -3883,22 +3883,23 @@ void Measure::checkTrailer()
 
 void Measure::setStretchedWidth(qreal w)
       {
-      qreal minWidth = score()->styleP(Sid::minMeasureWidth);
+      qreal minWidth = isMMRest() ? score()->styleP(Sid::minMMRestWidth) : score()->styleP(Sid::minMeasureWidth);
       if (w < minWidth)
             w = minWidth;
 
-      // multi measure rests are not stretched depending on their
-      // tick length
-
-      if (!isMMRest()) {
-            qreal stretchableWidth = 0.0;
-            for (Segment* s = first(SegmentType::ChordRest); s; s = s->next(SegmentType::ChordRest)) {
-                  if (!s->enabled())
-                        continue;
-                  stretchableWidth += s->width();
-                  }
-            w += stretchableWidth * (basicStretch()-1.0) * ticks().ticks() / 1920.0;
+      qreal stretchableWidth = 0.0;
+      for (Segment* s = first(SegmentType::ChordRest); s; s = s->next(SegmentType::ChordRest)) {
+            if (!s->enabled())
+                  continue;
+            stretchableWidth += s->width();
             }
+      int weight = ticks().ticks();
+      // reduce weight of mmrests
+      // so the nominal width is not directly proportional to duration
```

Comment:
Maybe.  I didn't want to change the current behavior too much because after discussions on Telegram and checking out previous forum discussions as well as Gould, we're actually in pretty good shape overall.  But for sure I'll test with some really long rests.

Reply:
Great call on the long rests, this change resulted in an mmrest taking up a whole system at less than 60 measures, whereas before you could have hundreds.

I'm going to cap the scaling, then, so you get different widths for different sizes up to the cap but then they are all the same.  This is one of those cases where we might want the *base* width to be the same for 100 measure versus 200, but we could let the 200 have greater weight during stretch, so in practice you would normally still see a difference in width.  I'm playing with solutions like this now.